\documentclass[11pt, oneside]{article}   	% use "amsart" instead of "article" for AMSLaTeX format
\usepackage[margin=0.75in]{geometry}                		% See geometry.pdf to learn the layout options. There are lots.
\geometry{letterpaper}                   		% ... or a4paper or a5paper or ... 
%\geometry{landscape}                		% Activate for rotated page geometry
%\usepackage[parfill]{parskip}    		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
\graphicspath{ {Figures/} }								% TeX will automatically convert eps --> pdf in pdflatex		
\usepackage{amssymb}
\usepackage[english]{babel}
\usepackage{csquotes}

\usepackage[backend=biber]{biblatex}
\addbibresource{basic.bib}

%SetFonts

%SetFonts


\title{Predicting TV Virulence: Employing Deep IV for Identifying Virulence Causative Genes}
\author{
	Jack Andraka\\ 
	\texttt{jandraka@stanford.edu}
	\and
	Billy Ferguson\\
	\texttt{billyf@stanford.edu}
	\and
	Charlie Walker\\
	\texttt{cwalker4@stanford.edu}
}
\date{}							% Activate to display a given date or no date

\begin{document}
\maketitle
\section{Introduction}

Despite the development of potentially curative chemotherapy, tuberculosis (TB) continues to cause increasing worldwide morbidity and is a leading cause of human mortality: 1.7 million individuals succumb to TB annually, and over 10 million more individuals contract the disease each year.\textsuperscript{[1]} The genome-wide association study (GWAS) is an experimental design used to detect associations between genetic variants and traits in samples from populations and has served as a main driver of understanding TB virulence.\textsuperscript{[2,3]} However, since traditional GWAS study designs run several thousand to million t-tests simultaneously they can only identify gene variants with large effect size. This is in contrast with the current hypothesis of genetic architecture that posits many gene variants acting in tandem to produce a phenotypic trait, each with small effect size.\textsuperscript{[4]} To address these shortcomings we will employ the Deep IV methodology on the Tuberculosis Gene Expression Dataset from the Khatri Lab to identify novel tuberculosis virulence-determining variants. 

\section{Methods}
We will implement the method described in ``Deep IV: A Flexible Approach for Counterfactual Prediction'' to identify the effect of gene expression on virulence of tuberculosis.\textsuperscript{[5]} GWA studies have historically been unable to recover significant relationships because of the endogeneity between gene expression and virulence and the enormous quantity of genes (on the order of ~4000). Instrumental variables (IV) are a well-developed tool for remedying endogeneity, but require a strong prior understanding of the data generating process and are not well equipped to deal with a large number of covariates. Deep IV promises to marry the best qualities of DNNs and IV, and we believe GWAS are a prime use case. \\

To perform IV analysis one needs to find an exogenous variable that affects the outcome variable only through the endogenous covariate of interest. More specifically, the instrument \emph{z} must be conditionally independent of the error (Figure 1).\\

Traditional IV can be estimated through a procedure called two-stage least squares (2SLS): in the first stage you regress your endogenous variable of interest, $p$,  on the exogenous instrument, $z$, to create a predicted $\hat{p}$ constructed only with the exogenous variation of $z$. In the second stage, you regress your outcome variable, $y$, on the predicted $\hat{p}$ from the first stage.\footnote{Chapter 4 of Angrist and Pischke's \emph{Mostly Harmless Econometrics: an Empiricist's Companion} provides a good introduction to instrumental variables.}
 \\
 
 
Figure 2 shows the general IV formulation applied to our question. As described in the introduction, the effect of gene expression on observed outcomes (in our case, virulence) is of enormous interest, but the direct estimation of these effects has thus far been unsuccessful. By instrumenting for gene expression with mutations in transcription factors we hope to more accurately predict virulence, as well as develop a more robust understanding of causal effects.\\

\begin{figure}[h]
\caption{Generalized Deep IV}
\centering
\includegraphics[width=0.6\textwidth]{figure_1.png}
\end{figure}


\begin{figure}[h]
\caption{Deep IV Application}
\centering
\includegraphics[width=0.6\textwidth]{figure_2.png}
\end{figure} 

\section{Challenges}
Given that the Deep IV methodology was just developed this past year, there are no other published articles performing this analysis. The Hartford et al. (2017) paper will therefore be our only reference describing the method and we will have to carry out the application of this mostly theoretical research. 

\pagebreak

\section{References}

\noindent [1] World Health Organization. \emph{Global Tuberculosis Report 2017}. 2017. Print.\\

\noindent [2] Uren, Caitlin, et al. ``A Post-GWAS Analysis of Predicted Regulatory Variants and Tuberculosis Susceptibility.'' \emph{PLoS One}, vol. 12, no. 4, June 2017, doi:10.1371/journal.pone.0174738.\\

\noindent [3] Bermingham, M L, et al. ``Genome-Wide Association Study Identifies Novel Loci Associated with Resistance to Bovine Tuberculosis.'' \emph{Heredity}, vol. 112, no. 5, May 2014, pp. 543?551., doi:10.1038/hdy.2013.137.\\

\noindent [4] Korte, Arthur, and Ashley Farlow. ``The Advantages and Limitations of Trait Analysis with GWAS: a Review.'' \emph{Plant Methods}, vol. 9, no. 29, 22 July 2013.\\

\noindent [5] Hartford, Jason, et al. ``Deep IV: A Flexible Approach for Counterfactual Prediction.'' \emph{International Conference on Machine Learning}. 2017.



\end{document}  